app/user.go

```diff
@@ -0,0 +1,597 @@
+	user, err := c.getUser(userID, userEmail)
+	if err != nil {
+		return nil, nil, err
+	}
+	var pageToken string
+	var roles []*auth.Role
+	for {
+		res, err := c.client.GetRoles(c.ctx, &authservice.GetRolesRequest{
+			PageToken: pageToken,
+			UserId:    user.Id,
+		})
+		if err != nil {
+			return nil, nil, err
+		}
+		roles = append(roles, res.Roles...)
+		// Check if we should continue paging
+		pageToken = res.NextPageToken
+		if len(pageToken) == 0 {
+			break
+		}
+	}
+
+	return user, roles, nil
+}
+
+func (c *UserClient) inviteUsers(
+	ctx *cli.Context,
+	emails []string,
+	namespacePermissions []string,
+	accountRole string,
+) error {
+	if len(accountRole) == 0 {
+		return errors.New("account role required for inviting new users")
+	}
+
+	// the role ids to invite the users for
+	var roleIDs []string
+
+	// first get the required account role
+	role, err := getAccountRole(c.ctx, c.client, accountRole)
+	if err != nil {
+		return err
+	}
+	roleIDs = append(roleIDs, role.GetId())
+
+	// get any optional namespace permissions
+	if len(namespacePermissions) > 0 {
+		npm, err := toUserNamespacePermissionsMap(namespacePermissions)
+		if err != nil {
+			return err
+		}
+		nsRoles, err := getNamespaceRolesFromMap(c.ctx, c.client, npm)
+		if err != nil {
+			return err
+		}
+		for _, nsRole := range nsRoles {
+			roleIDs = append(roleIDs, nsRole.GetId())
+		}
+	}
+
+	// Invite users to given roles
+	req := &authservice.InviteUsersRequest{
+		Specs:     make([]*auth.UserSpec, len(emails)),
+		RequestId: ctx.String(RequestIDFlagName),
+	}
+	for i := range emails {
+		req.Specs[i] = &auth.UserSpec{
+			Email: emails[i],
+			Roles: roleIDs,
+		}
+	}
+	resp, err := c.client.InviteUsers(c.ctx, req)
+	if err != nil {
+		return fmt.Errorf("unable to invite users: %w", err)
+	}
+	return PrintProto(resp.GetRequestStatus())
+}
+
+func (c *UserClient) resendInvitation(
+	ctx *cli.Context,
+	userID string,
+	userEmail string,
+) error {
+	user, err := c.getUser(userID, userEmail)
+	if err != nil {
+		return err
+	}
+	req := &authservice.ResendUserInviteRequest{
+		UserId:    user.Id,
+		RequestId: ctx.String(RequestIDFlagName),
+	}
+	resp, err := c.client.ResendUserInvite(c.ctx, req)
+	if err != nil {
+		return fmt.Errorf("unable to resend invitation for user: %w", err)
+	}
+	return PrintProto(resp.GetRequestStatus())
+}
+
+func (c *UserClient) deleteUser(
+	ctx *cli.Context,
+	userID string,
+	userEmail string,
+) error {
+	if userID != "" && userEmail != "" {
+		return fmt.Errorf("both user-id and user-email set")
+	}
+	if userID == "" && userEmail == "" {
+		return fmt.Errorf("both user-id and user-email cannot be empty")
+	}
+
+	res, err := c.client.GetUser(c.ctx, &authservice.GetUserRequest{
+		UserEmail: userEmail,
+		UserId:    userID,
+	})
+	if err != nil {
+		return err
+	}
+	req := &authservice.DeleteUserRequest{
+		UserId:          res.User.Id,
+		ResourceVersion: ctx.String(ResourceVersionFlagName),
```

Comment:
Should `ResourceVersionFlagName` be required since we use it directly in this command?

Reply:
its optional given we populate it from the existing user if the user doesn't supply a specific version